packages/frontend/src/analytics.ts

```diff
@@ -7,37 +7,147 @@ let applicationInsights: ApplicationInsights | null = null;
 
 const applicationInsightsEnabled = config.applicationInsightsInstrumentationKey && import.meta.env.PROD;
 
+// Enhanced page name mapping for better funnel analysis
+const getPageNameFromPath = (pathname: string): string => {
+  // Remove query parameters and fragments
+  const cleanPath = pathname.split('?')[0].split('#')[0];
+
+  // Map paths to meaningful page names
+  const pageMapping: Record<string, string> = {
+    '/': 'Inbox',
```

Comment:
Could we use PageRoutes enum here perhaps?

Reply:
aye!